### PR TITLE
Upgrade rubocop, motivated by CVE.

### PR DESCRIPTION
**Pull request by xaviershay**
Defaulted most new things to off, though tried to keep performance and linters.

Note that `.rubocop_rspec_base` doesn't 100% match up with `rspec-dev` because I was adding to it as we went along. Once these are all merged I'm planning to ensure they're all consistent.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -95,3 +89,22 @@ Style/RaiseArgs:
...
     - lib/rspec/core/option_parser.rb
     - lib/rspec/core/pending.rb
+
+Performance/RedundantBlockCall:
+  Exclude:
+    - lib/rspec/core/rake_task.rb
+
+Lint/IneffectiveAccessModifier:
+  Exclude:
+    - lib/rspec/core/memoized_helpers.rb # Fixing this file was too much of a diif
```
Comment (myronmarston) [inline .rubocop.yml]: Nit: "diif" is mispelled.

File: lib/rspec/core/formatters/deprecation_formatter.rb [hunk 2]
```diff
@@ -81,7 +81,7 @@ def too_many_warnings_message
 
           def output_formatted(str)
             return str unless str.lines.count > 1
-            separator = "#{'-' * 80}"
+            separator = '-' * 80
```
Comment (myronmarston) [inline lib/rspec/core/formatters/deprecation_formatter.rb]: Funny that we were wrapping this with string interpolation before...

File: lib/rspec/core/memoized_helpers.rb [hunk 3]
```diff
@@ -483,9 +483,9 @@ def subject!(name=nil, &block)
       def self.module_for(example_group)
         get_constant_or_yield(example_group, :LetDefinitions) do
           mod = Module.new do
-            include Module.new {
+            include(Module.new {
```
Comment (myronmarston) [inline lib/rspec/core/memoized_helpers.rb]: IMO, it's a little weird to use parens on `include` calls.  We don't normally do that.  Why here?

Comment (xaviershay) [reply lib/rspec/core/memoized_helpers.rb]: the rubocop warning was that the block was ambiguous:

```
some_method param { }
```

Does the block belong to the method or the param? That seemed legitimate enough to me to add brackets, but don't feel strongly.

**Comment by xaviershay** [comment]
Current build failure: timeout on ruby 2.4.3. Retried.

**Pull request merged**